core/actor/src/test/scala/net/liftweb/actor/LAFutureSpec.scala

```diff
@@ -61,30 +61,81 @@ class LAFutureSpec extends Specification {
       result shouldEqual givenFailure
     }
 
-    "collect one future result" in {
-      val givenOneResult = 123
-      val one = LAFuture(() => givenOneResult)
-      LAFuture.collect(one).get(timeout) shouldEqual List(givenOneResult)
-    }
+    "when collecting results with LAFuture.collect" in {
+      "collect one future result" in {
+        val givenOneResult = 123
+        val one = LAFuture(() => givenOneResult)
+        LAFuture.collect(one).get(timeout) shouldEqual List(givenOneResult)
+      }
 
-    "collect more future results in correct order" in {
-      val givenOneResult = 123
-      val givenTwoResult = 234
-      val one = LAFuture(() => givenOneResult)
-      val two = LAFuture(() => givenTwoResult)
-      LAFuture.collect(one, two).get(timeout) shouldEqual List(givenOneResult, givenTwoResult)
-    }
+      "collect more future results in correct order" in {
+        val givenOneResult = 123
+        val givenTwoResult = 234
+        val one = LAFuture(() => givenOneResult)
+        val two = LAFuture(() => givenTwoResult)
+        LAFuture.collect(one, two).get(timeout) shouldEqual List(givenOneResult, givenTwoResult)
+      }
+
+      "collect empty list immediately" in {
+        val collectResult = LAFuture.collect(Nil: _*)
+        collectResult.isSatisfied shouldEqual true
+        collectResult.get(timeout) shouldEqual Nil
+      }
 
-    "collect empty list immediately" in {
-      val collectResult = LAFuture.collect(Nil: _*)
-      collectResult.isSatisfied shouldEqual true
-      collectResult.get(timeout) shouldEqual Nil
+      "report a failed LAFuture as a failure for the overall future" in {
+        val one: LAFuture[Int] = new LAFuture
+        val two: LAFuture[Int] = LAFuture(() => 5)
+
+        one.fail(Failure("boom boom boom!"))
+
+        val collectResult = LAFuture.collect(one, two)
+        collectResult.get(timeout) shouldEqual Failure("boom boom boom!")
```

Comment:
It is not only weird, it's incorrect heh. The test above should be `shouldEqual Full(Nil)`.... Not entirely sure why that test is passing O.o